registry/handlers/tags.go

```diff
@@ -49,6 +51,43 @@ func (th *tagsHandler) GetTags(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	// do pagination if requested
+	q := r.URL.Query()
+	// get entries after latest, if any specified
+	if lastEntry := q.Get("last"); lastEntry != "" {
+		lastEntryIndex := sort.SearchStrings(tags, lastEntry)
+		tags = tags[lastEntryIndex+1:]
```

Comment:
According to the documentation for [`sort.SearchStrings`](https://golang.org/src/sort/search.go?s=3680:3724#L91), `x is not present (it could be len(a))`. This could lead to a panic here.  This should probably be covered in the unit tests and boundary tested.

Reply:
I just seem to remember having fixed that error before.
To do unittests, that would require I know a lot more regarding how unit tests are done in the handlers.

I propose making this change (fixing this), so it will not panic and then we can either merge and someone can make a PR for unit tests, or there can be made a PR based on this.

But considering this has been in the making since 2016, I am not having high hopes for anyone picking up the unittests. And not sure when I will get the time for it, sadly

Reply:
@eyJhb maybe at least to ensure no panic we can add something like
```go
if lastEntry := q.Get("last"); lastEntry != "" {
	lastEntryIndex := sort.SearchStrings(tags, lastEntry) + 1
	if lastEntryIndex < len(tags) {
		tags = tags[lastEntryIndex:]
	}
}
```
?

Reply:
@jdolitsky ahh damn it, that is a lot prettier than what I just pushed... Lets totally go with that! Let me just rebase again :)

Reply:
EDIT: we can't do that actually, as we need to set it to a empty list if it has no matches. That is at least what I would think if a user specifies something that is not in the tags (return nothing). But we can do this 

What about the newest change?

```go
	if lastEntry := q.Get("last"); lastEntry != "" {
		lastEntryIndex := sort.SearchStrings(tags, lastEntry)

		// as`sort.SearchStrings` can return len(tags), if the
		// specified `lastEntry` is not found, we need to
		// ensure it does not panic when slicing.
		if lastEntryIndex < len(tags) {
			tags = []string{}
		} else {
			tags = tags[lastEntryIndex+1:]
		}
	}
```

Reply:
Yes, noticed that too. I agree. This lgtm

Reply:
I did however like yours a lot, but we end up having somewhat the same :) Perfect! Lets see if the others agree.
@dmcgowan will you resolve if you think this solution is acceptable?

Reply:
Ping @dmcgowan :)

Reply:
Ping again, hope it is okay @dmcgowan :)

@thaJeztah anything we can do to get this merged?